### PR TITLE
248, c++, gold

### PR DESCRIPTION
**Pull request by ryanchou-dev**
test drive

---
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: solutions/usaco-647.mdx [hunk 1]
```diff
@@ -2,22 +2,68 @@
...
+using namespace std;
+
+int main() {
+	freopen("248.in", "r", stdin);
+
+	int n;
+	cin >> n;
+	vector<int> a(n);
+
+	int bst = 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-647.mdx:33]: ryan i'm begging you stop removing vowels from variable names

File: solutions/usaco-647.mdx [hunk 2]
```diff
@@ -2,22 +2,68 @@
...
+	for (int i = 0; i < n; i++) {
+		cin >> a[i];
+		bst = max(bst, a[i]);
+	}
+
+	vector<vector<int>> dp(n, vector<int>(n, -1));
```
Comment (SansPapyrus683) [inline solutions/usaco-647.mdx:39]: comment for what this represents

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-647.mdx (hunk 1 above)
Comment (ryanchou-dev) [reply solutions/usaco-647.mdx:33]: ITS TOO LATE IM ADDICTED

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/usaco-647.mdx [hunk 3]
```diff
@@ -2,22 +2,69 @@
...
+				dp[l][r] = a[l];
+				continue;
+			}
+
+			// merge prefix and suffix
+			for (int rr = l; rr < r; rr++) {
+				if (dp[l][rr] != -1 && dp[l][rr] == dp[rr + 1][r]) {
+					dp[l][r] = max(
+							dp[l][r],
+							dp[l][rr] + 1);
```
Comment (ChuangSheep) [inline solutions/usaco-647.mdx:56]:
```suggestion
							dp[l][rr] + 1
					);
```

**Review by ChuangSheep — APPROVED** [review]

**Pull request merged**